File: behavior/pattern/src/entities/Duck.cpp
#include <iostream>
#include "../interfaces/FlyBehavior.h"
#include "../interfaces/QuackBehavior.h"
using namespace std;

class Duck{

	FlyBehavior flyBehavior;
	QuackBehavior quackBehavior;

	public:
		void swin();
		void display();
		void performQuack();
		void performFly();
};

void Duck::swin(){
	cout << "Nadando" << endl;
}

void Duck::display(){
	cout << "Aparecendo" << endl;
}

void Duck::performQuack(){
	quackBehavior.quack();
}

void Duck::performFly(){
	flyBehavior.fly();
}

File: java/behavior/strategy/src/entities/Duck.java
package java.behavior.strategy.src.entities;
public class Duck implements IDuck{
    public FlyBehavior flyBehavior;
    public QuackBehavior quackBehavior;

    public Duck(FlyBehavior fb, QuackBehavior qb){
        flyBehavior = fb;
        quackBehavior = qb;
    }

    public void performFly(){
        flyBehavior.fly();
    }

    public void performQuack(){
        quackBehavior.quack();
    }

    public void setFlyBehavior(FlyBehavior fb){
        flyBehavior = fb;
    }

    public void setQuackBehavior(QuackBehavior qb){
        quackBehavior = qb;
    }

    @Override
    public void display(){
        System.out.println("Hi, im a normal duck");
    }

    @Override
    public void swin(){
        System.out.println("Swimming");
    }
}